server/hotrod/src/test/java/org/infinispan/server/hotrod/counter/StrongCounterAPITest.java

```diff
@@ -0,0 +1,111 @@
+package org.infinispan.server.hotrod.counter;
+
+import static org.infinispan.counter.api.CounterConfiguration.builder;
+import static org.infinispan.test.Exceptions.expectException;
+import static org.testng.Assert.assertNotEquals;
+import static org.testng.AssertJUnit.assertEquals;
+import static org.testng.AssertJUnit.assertFalse;
+import static org.testng.AssertJUnit.assertTrue;
+
+import java.lang.reflect.Method;
+import java.util.Random;
+
+import org.infinispan.configuration.cache.ConfigurationBuilder;
+import org.infinispan.counter.SyncStrongCounter;
+import org.infinispan.counter.api.CounterType;
+import org.infinispan.counter.exception.CounterOutOfBoundsException;
+import org.infinispan.server.hotrod.Constants;
+import org.infinispan.server.hotrod.HotRodMultiNodeTest;
+import org.testng.annotations.Test;
+
+/**
+ * A {@link org.infinispan.counter.api.StrongCounter} api test.
+ *
+ * @author Pedro Ruivo
+ * @since 9.2
+ */
+@Test(groups = "functional", testName = "server.hotrod.counter.StrongCounterAPITest")
+public class StrongCounterAPITest extends HotRodMultiNodeTest {
```

Comment:
I am not that familiar with client vs. server testsuite, but could you have a common base for tests in client & server? I am not fond of duplicities and TestCounters with non-trivial code just for the sake of tests either... @mgencur Do you know if this has been discussed before?

Reply:
No. I don't think the relation between hotrod-client and hotrod-server test suite has been discussed. I'd agree that when the tests are very similar they should have a common base class in the hotrod-server module.

Reply:
I don't think there are any common base test suite but you're right (at least for counters). The tests are the same and it is only testing the protocol (if everything is encoded/decoded correctly).

Reply:
I managed to share the test between server and client. I don't think it would won a beauty contest but it seems to work :)